Automattic/simplenote-macos
# Refreshing Notes List: Mark 3

### Description
In this PR we're restoring **Checklist Rendering Support** in the Notes List.

#### Why is this so tricky?:
The tooling we've got was meant to render **just one checklist item** per line of text. 

Since PR #541 we're now displaying "extended content" in the Notes List, which may include multiple lines (and each one may have its own List Marker)

#### Important Note
[This is not new code: iOS parent issue here](https://github.com/Automattic/simplenote-ios/pull/664), already in production!

@guarani Sir, may I trouble you with a last PR? (promise you, last one!!)
**Thanks in advance!!**

Ref. #458

### Test
1. Sign into your account
2. Add a new Note
3. Insert a checklist and add multiple lines

- [ ] Verify the new Checklist Items appear in the Notes List, in real time, as you type
- [ ] Verify the Notes List reacts well to Dark Mode
- [ ] Verify Increasing / Reducing the font size (CMD + / CMD -) causes the Editor to react properly
- [ ] Verify the Unit Tests are green

### Release
These changes do not require release notes.
